lib/ansible/modules/packaging/os/pkgutil.py

```diff
@@ -10,88 +10,140 @@
-      - The upgrade (C(latest)) operation will update/install the package to the latest version available.
-      - "Note: The module has a limitation that (C(latest)) only works for one package, not lists of them."
-    required: true
-    choices: ["present", "absent", "latest"]
+    - Whether to install (C(present)), or remove (C(absent)) packages.
+    - The upgrade (C(latest)) operation will update/install the packages to the latest version available.
+    type: str
+    choices: [ absent, installed, latest, present, removed ]
+    default: present
   update_catalog:
     description:
-      - If you want to refresh your catalog from the mirror, set this to (C(yes)).
-    required: false
-    default: False
+    - If you want to refresh your catalog from the mirror, set this to (C(yes)).
+    type: bool
+    default: no
+    version_added: '2.1'
+  force:
+    description:
+    - Allows the update process to downgrade packages to what is present in the repostory set this to (C(yes)).
+    - It is useful as a rollback to stable from testing, and similar operations.
     type: bool
-    version_added: "2.1"
+    default: no
+    version_added: '2.8'
 '''
 
-EXAMPLES = '''
-# Install a package
-- pkgutil:
+EXAMPLES = r'''
+- name: Install a package
+  pkgutil:
     name: CSWcommon
     state: present
 
-# Install a package from a specific repository
-- pkgutil:
+- name: Install a package from a specific repository
+  pkgutil:
     name: CSWnrpe
-    site: 'ftp://myinternal.repo/opencsw/kiel'
+    site: ftp://myinternal.repo/opencsw/kiel
+    state: latest
+
+- name: Remove a package
+  pkgutil:
+    name: CSWtop
+    state: absent
+
+- name: Install several packages
+  pkgutil:
+    name:
+    - CSWsudo
+    - CSWtop
+    state: present
+
+- name: Update all packages
+  pkgutil:
+    name: '*'
     state: latest
+
+- name: Update all packages and force versions to match latest in catalog
+  pkgutil:
+    name: '*'
+    state: latest
+    force: yes
 '''
 
+RETURN = r''' # '''
+
 from ansible.module_utils.basic import AnsibleModule
 
 
-def package_installed(module, name):
-    cmd = ['pkginfo']
-    cmd.append('-q')
-    cmd.append(name)
-    rc, out, err = run_command(module, cmd)
-    if rc == 0:
-        return True
-    else:
-        return False
+def packages_not_installed(module, names):
+    ''' Check if each package is installed and return list of the ones absent '''
+    pkgs = []
+    for pkg in names:
+        rc, out, err = run_command(module, ['pkginfo', '-q', pkg])
+        if rc != 0:
+            pkgs.append(pkg)
+    return pkgs
 
 
-def package_latest(module, name, site):
-    # Only supports one package
-    cmd = ['pkgutil', '-U', '--single', '-c']
+def packages_installed(module, names):
+    ''' Check if each package is installed and return list of the ones present '''
+    pkgs = []
+    for pkg in names:
+        if pkg.startswith('CSW'):
+            continue
```

Comment:
This looks like debugging code that should have been removed?  It causes the module to fail to detect installed OpenCSW packages.

Reply:
@mavit I assumed because there is a better tool to manage OpenCSW packages ?

Reply:
No, `pkgutil` is the OpenCSW package manager.  These are typically the only packages that it manages.

Reply:
Ok, I see what the problem is, it should only consider CSW packages.
I'll fix.

Reply:
Fixed.

Reply:
This leads to another issue.  The `pkgutil` command will quietly prefix package names with `CSW` if absent, and users might reasonably expect this Ansible module to behave in the same way.

Consider the following play, which will always return `changed`, `ok`, successfully installing the package but then neglecting to remove it.
```yaml
- pkgutil:
    name: top
    state: present

- pkgutil:
    name: top
    state: absent
```

It would be better to throw an error in this case rather than silently considering such packages never to be installed.

Reply:
Look, I can't fix it because I cannot test this code.

Reply:
(And mind you, I didn't wrote it either ;-))

Reply:
Well, I suppose you *could* install Solaris in a VM, but I do see why you’d consider than the problem of someone who cares enough about Solaris to already have some Solaris.  That’s cool.  Frankly I’m impressed you tried to take this on in the first place.

So, then, let’s merge what we have.  The tests all pass, now, and it’s a step in the right direction.

Reply:
Well, it usually starts small. And before you know it, it's refactored ;-)
If you are dedicated to test this module and improve the integration test, I prefer to merge and let you finish the job. Deal ?

Reply:
Deal.